.github/workflows/periodic-merge-kernel-updates-6h.yml

```diff
@@ -0,0 +1,41 @@
+# This action periodically merges base branches into batch-with-kernel-updates branch.
+# This is done to
+#  * prevent conflicts or rather resolve them early
+#  * make all potential breakage happen on the staging branch
+#  * and make sure that all major rebuilds happen before the staging
+#    branch get’s merged back into its base branch.
+
+name: "Periodic Merges into batch-with-kernel-updates (6h)"
+
+on:
+  schedule:
+    # * is a special character in YAML so you have to quote this string
+    # Merge every 6 hours
+    - cron: '0 */6 * * *'
+  workflow_dispatch:
+
+permissions: {}
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  periodic-merge:
+    if: github.repository_owner == 'NixOS'
+    strategy:
+      # don't fail fast, so that all pairs are tried
+      fail-fast: false
+      # certain branches need to be merged in order, like master->staging-next->staging
+      # and disabling parallelism ensures the order of the pairs below.
+      max-parallel: 1
+      matrix:
+        pairs:
+          - from: master
+            into: batch-with-kernel-updates
```

Comment:
Now that I think of it, maybe we should just add this pair to the other merge-6h jobset config?